src/pages/index.js

```diff
@@ -0,0 +1,307 @@
+    const modal = document.querySelector(".modal_opened");
+    closeModal(modal);
+  }
+}
+
+function modalOverlayClose(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+
+  setButtonText(submitButton, true, "Save", "Saving...");
+
+  const updatedUserInfo = {
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value,
+  };
+
+  api
+    .editUserInfo(updatedUserInfo)
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitButton.textContent = "Save";
+     // setButtonText(cardSubmitButton, false);
+    });
+} 
+
+profileEditButton.addEventListener("click", () => {
+  editModalNameInput.value = profileName.textContent;
+  editModalDescriptionInput.value = profileDescription.textContent;
+  resetValidation(editProfileFormElement, [editModalNameInput, editModalDescriptionInput]
+    , settings);
+  openModal(editModal);
+});
+
+closeButtons.forEach((button) => {
+  const popup = button.closest(".modal");
+  button.addEventListener('click', () => closeModal(popup));
+});
+
+cardModalButton.addEventListener("click", () => {
+  openModal(cardModal);
+});
+
+  editProfileFormElement.addEventListener("submit", handleEditFormSubmit);
+  cardFormElement.addEventListener("submit", handleAddCardSubmit);
+
+avatarModalButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+avatarModalCloseButton.addEventListener("click", () => {
+  closeModal(avatarModal);
+});
+
+avatarFormElement.addEventListener("submit", handleAvatarSubmit);
+
+deleteFormElement.addEventListener("submit", handleDeleteSubmit);
+
+const deleteModalCancelButton = deleteModal.querySelector(".modal__submit-button_cancel");
+deleteModalCancelButton.addEventListener("click", () => {
+  closeModal(deleteModal);
+});
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, "Create", "Creating...");
+
+  const inputValues = {
+    name: cardModalCaptionInput.value,
+    link: cardModalLinkInput.value
+  };
+
+  api
+    .createCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardFormElement.reset();
+      closeModal(cardModal);
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      setButtonText(submitButton, false, "Create", "Creating...");
+      const buttonElement = cardFormElement.querySelector(settings.submitButtonSelector);
+      disableButton(buttonElement, settings);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = avatarFormElement.querySelector(".modal__submit-button");
```

Comment:
Please, don't search the submit button here because in the other submit handlers, you have ` const submitButton = evt.submitter;`


```js

  const submitButton = evt.submitter;

  setButtonText(submitButton, true, "Save", "Saving...");
```